components/resourceDetails/resourceSharing/addAgentButton/addAgentRow/index.jsx

```diff
@@ -89,14 +90,21 @@ export default function AddAgentRow({
   const classes = useStyles();
   const [agentWebId, setAgentWebId] = useState("");
   const [existingPermission, setExistingPermission] = useState();
-
-  const agentName = (thing && getStringNoLocale(thing, foaf.name)) || null;
-  const agentAvatar = (thing && getUrl(thing, vcard.hasPhoto)) || null;
-  const displayedWebId =
-    (thing &&
-      contactsArrayLength > 0 &&
-      getUrl(thing, VCARD_WEBID_PREDICATE)) ||
-    null;
+  const [agentName, setAgentName] = useState(null);
+  const [agentAvatar, setAgentAvatar] = useState(null);
+  const [displayedWebId, setDisplayedWebId] = useState(null);
+  const { data: profile } = useContactProfile(thing);
+  useEffect(() => {
+    if (profile) {
+      setAgentName(profile.name);
+      setAgentAvatar(profile.avatar);
+      setDisplayedWebId(profile.webId);
+    } else if (thing && !profile && contactsArrayLength > 0) {
+      setAgentName(getStringNoLocale(thing, foaf.name) || null);
```

Comment:
This is in the case there is no profile but there is a thing, which means it's the Thing we created to which we added foaf.name on the fly so we know it's under foaf.name. If it's a contact without a profile it will set the name to null and we display the webId anyway like we do for all failed profiles.